plugins/modules/database/misc/redis.py

```diff
@@ -277,7 +278,10 @@ def main():
         name = module.params['name']
 
         try:  # try to parse the value as if it were the memory size
-            value = str(human_to_bytes(module.params['value'].upper()))
+            if re.match(r"^([\d\.])+.?B$", module.params['value'].upper()):
```

Comment:
This regex doesn't accept all valid forms. The regex used in `human_to_bytes` is `r'^\s*(\d*\.?\d*)\s*([A-Za-z]+)?'`, so here better use `r'^\s*(\d*\.?\d*)\s*([A-Za-z]+)?\s*$'`. That way you ensure that there is nothing following the size.